broken `std::optional<enum class : uint8_t>`

```
import cppyy
cppyy.cppdef("enum class X : uint8_t {P0,P1,P2}; std::optional<X> x;")
cppyy.gbl.x = cppyy.gbl.X.P1
```

will error with

```
E       TypeError: Template method resolution failed:
E         X std::optional<X>::value_or<X>(X && __u) =>
E           TypeError: could not convert argument 1: [InstanceMoveConverter]
```

whereas replacing `uint8_t` with `int` works fine.

This is exposes in a new xfail-ing test in https://github.com/compiler-research/cppyy/pull/209.

--- a/src/Converters.cxx
+++ b/src/Converters.cxx
@@ -3712,7 +3712,7 @@ CPyCppyy::Converter* CPyCppyy::CreateConverter(Cppyy::TCppType_t type, cdims_t d
                ) {
                 static STLIteratorConverter c;
                 result = &c;
-            } else {
+            } else if(realTypeStr != "int8_t" and realTypeStr != "uint8_t") {
        // -- Cling WORKAROUND
                 result = selectInstanceCnv(klass, cpd, dims, isConst, control);
             }

